Makefile

```diff
@@ -341,6 +347,10 @@ DC_TOP		:= $(STARSHIP_TOP)
 #
 #######################################
 
+sync:
```

Comment:
I suggest naming the target "fpga-path", and moving *.patch into a new directory instead of the root.

Reply:
要不这样，我建一个patch文件夹，所有patch方里面，然后里面加一个make负责给每个patch作apply。然后root文件夹就加一个preprocess任务，作同步和patch

Reply:
Is the sync necessary? I think only patch is enough.

Reply:
but I can save a line of "git submodule update" each time I clone it again. I think every users hope can run the total project by coding only one line of code.

Reply:
I prefer using `proxychains` to set up the proxy instead of using environment variables, therefore I feel uncomfortable to hardcode git commit in the scripts.
Besides, I plan to use another tool to manage dependencies in the future. Before that, you can add another target for these submodules.